### PR TITLE
fix(major): [sc-6337] Add dictonary compression support.

### PR DESCRIPTION
**Pull request by ser-0xff**
## Description

fix(major): [sc-6337] Add dictonary compression support.

## How Has This Been Tested?

Unit tests.

## Minimal checklist:

- [X] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Sources/DistributedSystem/ChannelCompression.swift [hunk 1]
```diff
@@ -67,23 +99,40 @@ final class ChannelCompressionHandshakeServer: ChannelInboundHandler, RemovableC
...
-        buffer.writeInteger(compression, as: UInt8.self)
-        context.writeAndFlush(NIOAny(buffer), promise: nil)
-        if compression != 0 {
-            _ = context.pipeline.addHandler(ChannelCompressionInboundHandler(loggerBox), position: .after(streamHandler))
-            _ = context.pipeline.addHandler(ChannelCompressionOutboundHandler(), position: .last)
+        let compressionMode = distributedSystem.compressionMode
+        switch compressionMode {
+        case .disabled:
+            var buffer = ByteBufferAllocator().buffer(capacity: MemoryLayout<UInt8>.size)
+            buffer.writeInteger(UInt8(0))
```
Comment (dimlio) [inline Sources/DistributedSystem/ChannelCompression.swift:115]: Can these magic 0, 1 and 2 can be defined once?

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 2]
```diff
@@ -1421,4 +1475,13 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
         }
         return try await future.get()
     }
+
+    func incrementStats(_ stats: [String: UInt64]) {
+        lock.withLock {
+            for (key, value) in stats {
+                let currentValue = self.stats[key, default: 0]
```
Comment (dimlio) [inline Sources/DistributedSystem/DistributedSystem.swift:1482]:
```suggestion
                self.stats[key, default: 0] += value
```

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 3]
```diff
@@ -25,6 +25,41 @@ extension Channel {
     }
 }
 
+public enum CompressionMode {
+    public struct Dictionary {
+        let data: UnsafeRawBufferPointer
+        let checksum: UInt32
+
+        // very simple checksum calculation
```
Comment (dimlio) [inline Sources/DistributedSystem/DistributedSystem.swift:33]: Maybe use e.g. CRC32, according to ChatGPT it has rather simple implementation:
```swift
import Foundation

/// Computes the CRC32 checksum for the given buffer.
/// - Parameter buffer: The data buffer for which the CRC32 checksum is computed.
/// - Returns: The CRC32 checksum as a `UInt32`.
func crc32(buffer: Data) -> UInt32 {
    let polynomial: UInt32 = 0xEDB88320
    var crc: UInt32 = 0xFFFFFFFF

    for byte in buffer {
        var currentByte = UInt32(byte)
        for _ in 0..<8 {
            let mix = (crc ^ currentByte) & 1
            crc >>= 1
            if mix != 0 {
                crc ^= polynomial
            }
            currentByte >>= 1
        }
    }

    return ~crc
}

// Example usage
let data = "Hello, World!".data(using: .utf8)!
let checksum = crc32(buffer: data)
print(String(format: "CRC32: %08X", checksum))
```

Verified it for some examples using https://crc32.online/ :)

**Review by dimlio — APPROVED** [review]

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 3 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift:33]: Your code works with `Data` type, should be adopted to UnsafeRawBufferPointer.
Let's leave as is for now :)

File: Sources/DistributedSystem/ChannelCompression.swift (hunk 1 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/ChannelCompression.swift]: Fixed.

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 2 above)
Comment (ser-0xff) [reply Sources/DistributedSystem/DistributedSystem.swift]: Fixed.

File: Sources/DistributedSystem/DistributedSystem.swift (hunk 3 above)
Comment (dimlio) [reply Sources/DistributedSystem/DistributedSystem.swift:33]:
```swift
import Foundation

/// Computes the CRC32 checksum for the given buffer.
/// - Parameter buffer: The raw buffer for which the CRC32 checksum is computed.
/// - Returns: The CRC32 checksum as a `UInt32`.
func crc32(buffer: UnsafeRawBufferPointer) -> UInt32 {
    let polynomial: UInt32 = 0xEDB88320
    var crc: UInt32 = 0xFFFFFFFF

    for byte in buffer {
        var currentByte = UInt32(byte)
        for _ in 0..<8 {
            let mix = (crc ^ currentByte) & 1
            crc >>= 1
            if mix != 0 {
                crc ^= polynomial
            }
            currentByte >>= 1
        }
    }

    return ~crc
}

// Example usage
let string = "Hello, World!"
string.withCString { cString in
    let length = strlen(cString)
    let buffer = UnsafeRawBufferPointer(start: cString, count: length)
    let checksum = crc32(buffer: buffer)
    print(String(format: "CRC32: %08X", checksum)) // Output: CRC32: 1C291CA3
}
```

Enjoy!

**Pull request merged**